spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,273 @@
+			}
+		}
+		Object result;
+		try {
+			result = method.invoke(bean, arguments);
+		}
+		catch (Exception e) {
+			throw new BeanInitializationException("Cannot setup StreamEmitter for " + method, e);
+		}
+
+		if (!Void.TYPE.equals(method.getReturnType())) {
+			if (targetBean == null) {
+				targetBean = this.applicationContext.getBean(outboundName);
+			}
+			boolean streamListenerResultAdapterFound = false;
+			for (StreamListenerResultAdapter streamListenerResultAdapter : this.streamListenerResultAdapters) {
+				if (streamListenerResultAdapter.supports(result.getClass(), targetBean.getClass())) {
+					Object adapted = streamListenerResultAdapter.adapt(result, targetBean);
+					if (adapted != null && Disposable.class.isAssignableFrom(adapted.getClass())) {
+						fluxDisposables.add((Disposable) adapted);
+					}
+					streamListenerResultAdapterFound = true;
+					break;
+				}
+			}
+			Assert.state(streamListenerResultAdapterFound, StreamEmitterErrorMessages.CANNOT_CONVERT_RETURN_TYPE_TO_ANY_AVAILABLE_RESULT_ADAPTERS);
+		}
+	}
+
+	private static void validateStreamEmitterMethod(Method method, int outputAnnotationCount, String methodAnnotatedOutboundName) {
+		if (StringUtils.hasText(methodAnnotatedOutboundName)) {
+			Assert.isTrue(outputAnnotationCount == 0,
+					StreamEmitterErrorMessages.INVALID_OUTPUT_METHOD_PARAMETERS);
+		}
+		else {
+			Assert.isTrue(outputAnnotationCount > 0, StreamEmitterErrorMessages.NO_OUTPUT_SPECIFIED);
+		}
+
+		if (!method.getReturnType().equals(Void.TYPE)) {
+			Assert.isTrue(StringUtils.hasText(methodAnnotatedOutboundName),
+					StreamEmitterErrorMessages.RETURN_TYPE_NO_OUTBOUND_SPECIFIED);
+			Assert.isTrue(method.getParameterCount() == 0, StreamEmitterErrorMessages.RETURN_TYPE_METHOD_ARGUMENTS);
+		}
+		else {
+			if (!StringUtils.hasText(methodAnnotatedOutboundName)) {
+				int methodArgumentsLength = method.getParameterTypes().length;
+				for (int parameterIndex = 0; parameterIndex < methodArgumentsLength; parameterIndex++) {
+					MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+					if (methodParameter.hasParameterAnnotation(Output.class)) {
+						String outboundName = (String) AnnotationUtils
+								.getValue(methodParameter.getParameterAnnotation(Output.class));
+						Assert.isTrue(StringUtils.hasText(outboundName), StreamEmitterErrorMessages.INVALID_OUTBOUND_NAME);
+					}
+					else {
+						throw new IllegalArgumentException(
+								StreamEmitterErrorMessages.OUTPUT_ANNOTATION_MISSING_ON_METHOD_PARAMETERS_VOID_RETURN_TYPE);
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return true;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		if (callback != null) {
+			callback.run();
+		}
+	}
+
+	@Override
+	public void stop() {
+		try {
+			lock.lock();
+			for (FluxSender fluxSender : fluxSenders) {
+				if (fluxSender.getClass().isAssignableFrom(FluxSenderImpl.class)) {
```

Comment:
Fine with me, too. 
And with that we may come back to the anonymous implementation in that ParameterAdapter.
And already without that ugly `...Impl` name. 

I've chosen `FluxSenderImpl` just as a quick sample about target implementation for the interface. The name really was just an arbitrary. You call name it with other proper name. If we will stick with that yet, of course...